state_transfer/route_endpoint.go

```diff
@@ -0,0 +1,124 @@
+package state_transfer
+
+import (
+	"context"
+	"reflect"
+
+	routev1 "github.com/openshift/api/route/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type RouteEndpoint struct {
+	hostname string
+	port     int32
+}
+
+func (r *RouteEndpoint) createEndpointResources(c client.Client, t Transfer) error {
+	err := createRouteService(c, r, t)
+	if err != nil {
+		return err
+	}
+
+	err = createRoute(c, r, t)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createRouteService(c client.Client, r *RouteEndpoint, t Transfer) error {
+	serviceSelector := labels
+	serviceSelector["pvc"] = t.PVC().Name
+
+	service := v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      t.PVC().Name,
+			Namespace: t.PVC().Namespace,
+			Labels:    labels,
+		},
+		Spec: v1.ServiceSpec{
+			Ports: []v1.ServicePort{
+				{
+					Name:       t.PVC().Name,
+					Protocol:   v1.ProtocolTCP,
+					Port:       t.Transport().Port(),
+					TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: t.Transport().Port()},
+				},
+			},
+			Selector: serviceSelector,
+			Type:     v1.ServiceTypeClusterIP,
+		},
+	}
+
+	return c.Create(context.TODO(), &service, &client.CreateOptions{})
+}
+
+func createRoute(c client.Client, r *RouteEndpoint, t Transfer) error {
+	termination := &routev1.TLSConfig{}
+	switch reflect.TypeOf(t.Transport()) {
+	case reflect.TypeOf(&NullTransport{}):
```

Comment:
question for my own understanding: I know that routes only work with `https` so with this config the transport is https?

Reply:
ughh well http? since it has the insecure edge termination policy?

Reply:
Routes work with http and https. Rclone without stunnel is just http:// on port 80. If the transport is null there is no ssl, hence http. 

Yes this is insecure. I called it out in the matrix in the README:
https://github.com/konveyor/crane-lib/blob/5529834f166d3d4555a137e23897e18402a51411/state_transfer/README.md

Reply:
Routes (haproxy) support:
```
HTTP
HTTPS (with SNI)
WebSockets
TLS with SNI
```

They do not support other tcp connections (like rsync when it's not wrapped in stunnel). Perhaps that is the confusion.